### PR TITLE
feat(stackRotate): Add stack rotate tool

### PR DESCRIPTION
**Pull request by m00n620**
StackRoateTool is implemented to rotate images on stack viewport. @sedghi , please take a look.

### CODE REVIEW

File: packages/tools/examples/stackManipulationTools/index.ts [hunk 1]
```diff
@@ -68,14 +68,14 @@ async function run() {
...
-  toolGroup.addTool(WindowLevelTool.toolName);
+  toolGroup.addTool(StackRotateTool.toolName);
   toolGroup.addTool(PanTool.toolName);
   toolGroup.addTool(ZoomTool.toolName);
   toolGroup.addTool(StackScrollMouseWheelTool.toolName);
 
   // Set the initial state of the tools, here all tools are active and bound to
   // Different mouse inputs
-  toolGroup.setToolActive(WindowLevelTool.toolName, {
+  toolGroup.setToolActive(StackRotateTool.toolName, {
```
Comment (sedghi) [inline packages/tools/examples/stackManipulationTools/index.ts:78]: can you not make that active tool by default here? window level should be the active tool

File: packages/tools/examples/stackRotateTool/index.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
+import { RenderingEngine, Types, Enums } from '@cornerstonejs/core';
```
Comment (sedghi) [inline packages/tools/examples/stackRotateTool/index.ts:1]: I didn't mean to create a new example. I was talking about just having it in the stack manipulation tools, but don't make it active by default.

Comment (m00n620) [reply packages/tools/examples/stackRotateTool/index.ts]: Resolved, please check again. @sedghi

**Comment by m00n620** [comment]
@lscoder, can you please review this as soon as possible?

**Comment by lscoder** [comment]
It stops working when I move the cursor to the center of the viewport (no error logged on console).

stackManipulation

**Comment by m00n620** [comment]
> It stops working when I move the cursor to the center of the viewport (no error logged on console).
> 
> stackManipulation [ stackManipulation ](https://user-images.githubusercontent.com/931820/209190692-2a2435e3-6574-45da-9e7e-876f755297d9.gif) [ ](https://user-images.githubusercontent.com/931820/209190692-2a2435e3-6574-45da-9e7e-876f755297d9.gif)

It's weird, I cannot reproduce that on my side.

**Comment by m00n620** [comment]
@lscoder , please check again. It should be working now.

**Comment by lscoder** [comment]
> @lscoder , please check again. It should be working now.

The issue is fixed and it is working fine now!

I pushed some changes to [lscoder/pr329](https://github.com/lscoder/cornerstone3D-beta/tree/lscoder/pr329) that update the example page adding a dropdown to select the left mouse button tool. Before users would have to change the code to be able to test the StackRotate tool. You can cherry-pick that commit to your branch.

```shell
$ git checkout feat/stack-rotate-tool
$ git remote add lscoder git@github.com:lscoder/cornerstone3D-beta.git
$ git fetch lscoder
$ git cherry-pick 7a7aded5
```
Or fork [lscoder/cornerstone3D-beta](git@github.com:lscoder/cornerstone3D-beta.git) in a new folder, create a patch for `7a7aded5` commit and apply it to your branch. Once you update the PR I will approve it to get it merged.

**Comment by m00n620** [comment]
> @m00n620 Can you do what is explained here for API check? https://www.cornerstonejs.org/docs/contribute/update-api/
> 
> We can merge after that

@sedghi , done, please take a look

**Review by sedghi — APPROVED** [review]

**Pull request merged**